### PR TITLE
HUD: do not obfuscate weapons/ammo in compact hud (4) while tracking

### PR DESCRIPTION
**Pull request by ciscon**

### CODE REVIEW

**Comment by ciscon** [comment]
added fix for lto/o3+ compilation/linking

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ else
 
     # Resolve all symbols at link time
     ifeq ($(SYS),Linux)
-    	CFLAGS += -DX11_GAMMA_WORKAROUND
+        CFLAGS += -DX11_GAMMA_WORKAROUND -fno-tree-vectorize
```
Comment (jite) [inline Makefile]: Please keep this in a separate commit. Not sure we should merge it tbh. What you could do it keep a local .config file where you add it to your CFLAGS for builds with higher optimization levels. (O3 is rarely recommended due to many potential bugs, is there a reason to why you want to compile with O3?)

File: vx_stuff.c [hunk 2]
```diff
@@ -91,7 +91,7 @@ cvar_t		amf_part_teleport = {"gl_particle_telesplash", "0"};
 cvar_t		amf_part_fire = {"gl_particle_fire", "0"}; // 1
 cvar_t		amf_part_firecolor = {"gl_particle_firecolor", ""}; // 1
 cvar_t		amf_part_sparks = {"gl_particle_sparks", "0"}; // 1
-cvar_t		amf_part_traillen    = {"gl_particle_trail_lenght", "1"};
+cvar_t		amf_part_traillen    = {"gl_particle_trail_length", "1"};
```
Comment (jite) [inline vx_stuff.c]: Could you make this (and the corresponding doc change) a separate commit too please. This will break configs as current configs have the misspelled cvar stored so it should be easy to spot in the git history. Thanks

File: gl_rmain.c [hunk 3]
```diff
@@ -217,6 +218,8 @@ cvar_t gl_modulate                         = {"gl_modulate", "1"};
 cvar_t gl_outline                          = {"gl_outline", "0"};
 cvar_t gl_outline_width                    = {"gl_outline_width", "2"};
 
+#define ISDEAD(i) ( (i) >= 41 && (i) <= 102 )
```
Comment (jite) [inline gl_rmain.c]: There's already 3 defines for this macro, lets not create a fourth :)

File: Makefile (hunk 1 above)
Comment (ciscon) [inline Makefile]: tree vectorization doesn't get enabled until O3 anyway so unless someone explicitly sets O3 on it shouldn't change anything.  I do get a small (somewhere around 5%) speed increase at O3, and this is the only optimization enabled by it that breaks anything.

File: gl_rmain.c (hunk 3 above)
Comment (ciscon) [inline gl_rmain.c]: Understandable, how would you like me to handle it?

**Comment by ciscon** [comment]
separated out commits
